plugin/documents.py

```diff
@@ -511,6 +512,21 @@ def on_query_completions(self, prefix: str, locations: List[int]) -> Optional[su
             lambda: self._on_query_completions_async(completion_list, locations[0], triggered_manually))
         return completion_list
 
+    def _format_on_paste(self):
+        self.purge_changes_async()
+        clipboard_text = sublime.get_clipboard()
+        sel = self.view.sel()
+        number_of_cursors = len(sel)
+        split_clipboard_text = clipboard_text.split('\n')
+        for index, region in enumerate(sel):
+            look_text = clipboard_text
+            if len(split_clipboard_text) == number_of_cursors:
+                look_text = split_clipboard_text[index]
+            found_region = self.view.find(look_text, region.end(), sublime.REVERSE)
+            if found_region:
+                sel.add(found_region)
+        self.view.run_command('lsp_format_document_range')
```

Comment:
This will leave the pasted content selected, I believe. Normal paste doesn't do that.

Reply:
[7e1b2d8](https://github.com/sublimelsp/LSP/pull/2311/commits/7e1b2d8205dfccce45f11ca0b45906b8ea1149da)